src/Microsoft.Health.Dicom.Core.UnitTests/Features/CustomTag/CustomTagCacheTests.cs

```diff
@@ -0,0 +1,98 @@
+using System.Threading.Tasks;
+using Microsoft.Health.Dicom.Core.Features.CustomTag;
+using NSubstitute;
+using Xunit;
+
+namespace Microsoft.Health.Dicom.Core.UnitTests.Features.CustomTag
+{
+    public class CustomTagCacheTests
+    {
+        private static readonly TimeSpan DefaultExpirationInterval = TimeSpan.FromSeconds(30);
+
+        [Fact]
+        public async Task GivenUninitializedCustomTagCache_WhenGetCustomTagsIsCalled_ShouldInitialize()
+        {
+            long expectedVersion = 1;
+            List<CustomTagEntry> expectedResult = new List<CustomTagEntry>();
+            expectedResult.Add(new CustomTagEntry() { Version = expectedVersion });
+            ICustomTagStore customTagStore = CreateCustomTagStore(expectedResult, expectedResult);
+            CustomTagCache customTagCache = new CustomTagCache(customTagStore, DefaultExpirationInterval);
+            CustomTagList customTags = await customTagCache.GetCustomTagsAsync();
+            Assert.Equal(expectedResult, customTags.CustomTags);
+            Assert.NotNull(customTagCache.LastRefreshTime);
+        }
+
+        [Fact]
+        public async Task GivenOutdatedCustomTagCache_WhenGetCustomTagsIsCalled_ShouldRefreshCache()
+        {
+            long expectedVersion1 = 1, expectedVersion2 = 2;
+            List<CustomTagEntry> expectedResult1 = new List<CustomTagEntry>() { CustomTagTestHelper.CreateCustomTagEntry(version: expectedVersion1) };
+            List<CustomTagEntry> expectedResult2 = new List<CustomTagEntry>() { CustomTagTestHelper.CreateCustomTagEntry(version: expectedVersion2) };
+            ICustomTagStore customTagStore = CreateCustomTagStore(expectedResult1, expectedResult2);
+
+            CustomTagCache customTagCache = new CustomTagCache(customTagStore, TimeSpan.FromSeconds(1));
+            CustomTagList customTags1 = await customTagCache.GetCustomTagsAsync();
+            DateTimeOffset lastRefreshTime = customTagCache.LastRefreshTime.Value;
+            await Task.Delay(TimeSpan.FromSeconds(2));
+
+            CustomTagList customTags2 = await customTagCache.GetCustomTagsAsync();
+            Assert.Equal(expectedResult2, customTags2.CustomTags);
+            Assert.NotEqual(lastRefreshTime, customTagCache.LastRefreshTime);
+        }
+
+        [Fact]
+        public async Task GivenRefreshedCustomTagCache_WhenGetCustomTagsIsCalled_ShouldNotRefreshCache()
+        {
+            long expectedVersion1 = 1, expectedVersion2 = 2;
+            List<CustomTagEntry> expectedResult1 = new List<CustomTagEntry>() { CustomTagTestHelper.CreateCustomTagEntry(expectedVersion1) };
+            List<CustomTagEntry> expectedResult2 = new List<CustomTagEntry>() { CustomTagTestHelper.CreateCustomTagEntry(expectedVersion2) };
+            ICustomTagStore customTagStore = CreateCustomTagStore(expectedResult1, expectedResult2);
+
+            CustomTagCache customTagCache = new CustomTagCache(customTagStore, TimeSpan.FromSeconds(6));
```

Comment:
>TimeSpan.FromSeconds(6) [](start = 79, length = 23)

Any reason you aren't using default here?

Reply:
it can use default one. let me update